tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1406,3 +1406,69 @@ def test_verify_attribute_of_users_are_updated(session, ldap_data, ldap_tear_dow
         assert ldap_data['ldap_user_name'] in user_values['user']['firstname']
         assert ldap_data['ldap_user_name'] in user_values['user']['lastname']
         assert ldap_data['ldap_user_name'] in user_values['user']['mail']
+
+
+@tier2
+@run_in_one_thread
+def test_verify_ldap_filters_ipa(session, ipa_data, ldap_tear_down):
+    """Verifying ldap filters in authsource to restrict access
+
+    :id: 0052b272-08b1-11eb-80c6-0c7a158cbff4
+
+    :Steps:
+        1. Create authsource with onthefly enabled and ldap filter
+        2. Verify login from users according to the filter
+
+    :expectedresults: Login fails for restricted user
+
+    """
+    result = ssh.command(
+        cmd=f"echo {settings.ipa.password_ipa} | kinit admin", hostname=settings.ipa.hostname_ipa
+    )
+    assert result.return_code == 0
+    test_user = gen_string('alpha')
+    add_user_cmd = (
+        f"echo {settings.ipa.password_ipa} | ipa user-add {test_user} --first"
+        f"={test_user} --last={test_user} --password"
+    )
+    result = ssh.command(cmd=add_user_cmd, hostname=settings.ipa.hostname_ipa)
+    assert result.return_code == 0
+    ldap_auth_name = gen_string('alphanumeric')
+    with session:
+        session.ldapauthentication.create(
+            {
+                'ldap_server.name': ldap_auth_name,
+                'ldap_server.host': ipa_data['ldap_ipa_hostname'],
+                'ldap_server.server_type': LDAP_SERVER_TYPE['UI']['ipa'],
+                'account.account_name': ipa_data['ldap_ipa_user_name'],
+                'account.password': ipa_data['ldap_ipa_user_passwd'],
+                'account.base_dn': ipa_data['ipa_base_dn'],
+                'account.groups_base_dn': ipa_data['ipa_group_base_dn'],
+                'account.onthefly_register': True,
+                'attribute_mappings.login': LDAP_ATTR['login'],
+                'attribute_mappings.first_name': LDAP_ATTR['firstname'],
+                'attribute_mappings.last_name': LDAP_ATTR['surname'],
+                'attribute_mappings.mail': LDAP_ATTR['mail'],
+            }
+        )
+
+        # 'test_user' able to login before the filter is applied.
+        with Session(user=test_user, password=ipa_data['ldap_ipa_user_passwd']) as ldapsession:
+            # with raises(NavigationTriesExceeded):
+            #     ldapsession.usergroup.search('')
```

Comment:
In comment you said it was for debugging and you have removed that, but it's only commented out?
I don't think it makes sense to merge new commented out code.
FYI, you can use `git add -p` to selectively stage changes, i.e. you can commit only some changes in file, while others are still there locally. Good for cases where some part is needed for debugging and whatnot, but shouldn't reach production.

Reply:
Well I have to rebase the code after omkar's PR( #8045 ) gets merged. So I thought I'd remove it in the next commit.